File: 3025-find-the-number-of-ways-to-place-people-i/3025-find-the-number-of-ways-to-place-people-i.py
class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        ans = 0
        for i in range(len(points)):
            x1, y1 = points[i]
            for j in range(len(points)):
                if i == j: continue
                x2, y2 = points[j]
                if x1 >= x2 and y1 <= y2:
                    for k, p in enumerate(points):
                        if k == i or k == j: continue
                        if x1 >= p[0] >= x2 and y1 <= p[1] <= y2:
                            ans -= 1
                            break
                    ans += 1
        return ans

File: 3025-find-the-number-of-ways-to-place-people-i/3025-find-the-number-of-ways-to-place-people-i.cpp
class Solution {
public:
    int numberOfPairs(vector<vector<int>>& points) {
        int ans = 0;
        
        for(int i=0; i<points.size(); i++){
            for(int j=0; j<points.size(); j++){
                if(i == j) continue;
                
                if(points[j][0] >= points[i][0] && points[j][1] <= points[i][1]){
                    // vector<vector<int>> dirs = {points[i],points[j],
                    //                             {points[i][0], points[j][1]},{points[i][1], points[j][0]}};
                    bool flag = true;
                    for(int k=0; k<points.size(); k++){
                        if(i == k or j == k) continue;
                        if(points[i][0] <= points[k][0] && points[k][0] <= points[j][0]
                            && points[j][1] <= points[k][1] && points[k][1] <= points[i][1]){
                            flag = false;
                        }
                    }
                    if (flag) ans++;
                }
                
            }
        }
        return ans;
    }
};